src/pages/Poll.jsx

```diff
@@ -1,21 +1,79 @@
-import React, { useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom';
 import { PollStart } from './PollStart';
 import { PollTake } from './PollTake';
 import { PollResult } from './PollResult';
+import PollService from '../API/PollService';
+import { Spin, Space } from 'antd';
+import 'antd/dist/antd.css';
 
 const Poll = () => {
-  const [pollStage, setPollStage] = useState(0);
-
-  const updatePollStage = (value) => {
-    setPollStage(value);
+  const [poll, setPoll] = useState({
+    title: '',
+    questionsCount: 0,
+    deadline: '',
+    users: [],
+    respondentsCount: 0,
+    selectedUsers: [],
+    status: "",
+  });
+  const updatePollStatus = (value) => {
+    let newPoll = poll;
+    newPoll.status = value;
+    setPoll(newPoll);
   };
+  const [isPollLoading, setIsPollLoading] = useState(false);
+  const pollId = useParams().id;
+
+  const usersList = useMemo(() => {
+    return poll.users.map(({ id, name }) => ({ label: name, value: id }));
+  }, [poll]);
+
+  useEffect(() => {
+    getPoll();
+  }, []);
+
+  async function getPoll() {
+    setIsPollLoading(true);
+    const response = await PollService.getPoll(pollId);
+    const changedResponse = {
+      respondentsCount: response.respondents.length,
+      users: response.respondents.map((item) => ({
+        id: item.userId,
+        name: item.secondName + ' ' + item.firstName,
+      })),
+      selectedUsers: response.respondents.map((item) => item.userId),
+      title: response.title,
+      questionsCount: response.questionsCount,
+      deadline: response.deadline,
+      status: response.status,
+    };
+    setPoll(changedResponse);
+    setIsPollLoading(false);
+  }
 
   return (
     <div>
-      {pollStage == 0 ? (
-        <PollStart updatePollStage={updatePollStage} />
+      {isPollLoading ? (
+        <Space className="data-loader">
+          <Spin size="large" />
+        </Space>
       ) : (
-        <>{pollStage == 1 ? <PollTake updatePollStage={updatePollStage} /> : <PollResult />}</>
+        <>
+          {poll.status == 'OPEN' ? (
+            <PollStart 
+              poll={poll}
+              usersList={usersList}
+              updatePollStatus={updatePollStatus} />
+          ) : (
+            <>
+              {poll.status == 'PROGRESS' 
+                ? <PollTake 
+                    updatePollStatus={updatePollStatus} /> 
+                : <PollResult />}
+            </>
+          )}
```

Comment:
давай без 3-х этажных тернарок
```suggestion
if (poll.status == 'OPEN') {
    return (
        <PollStart 
              poll={poll}
              usersList={usersList}
              updatePollStatus={updatePollStatus} />
    )
}
if (poll.status === 'PROGRESS') {
       return  (
       <PollTake 
                    updatePollStatus={updatePollStatus} /> 
          )
}

return <PollResult />
```

Reply:
Согласен